rustup-init.sh

```diff
@@ -90,15 +90,32 @@ main() {
     local need_tty=yes
     for arg in "$@"; do
         case "$arg" in
-            -h|--help)
+            --help)
                 usage
                 exit 0
                 ;;
-            -y)
-                # user wants to skip the prompt -- we don't need /dev/tty
-                need_tty=no
-                ;;
             *)
+                OPTIND=1
+                if [ "${arg%%--*}" = "" ]; then
+                    # Long option (other than --help);
+                    # don't attempt to interpret it.
+                    continue
+                fi
+                while getopts :hy sub_arg "$arg"; do
```

Comment:
Can you confirm if this will work in combination with long options?  i.e. if we had `--profile=blah -y --default-toolchain=whatever` would that work?

Reply:
Yes. The above logic will skip those.

```
if [ "${arg%%--*}" = "" ]; then
    # Long option (other than --help);
    # don't attempt to interpret it.
    continue
fi
```

The test `if [ "${arg%%--*}" = "" ]` simply checks whether the first two characters of the argument are `--`, and skips checking it if so.